metrics/metrics.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"k8s.io/client-go/util/workqueue"
+)
+
+var tagName = tag.MustNewKey("name")
+
+type intMetric struct {
+	name     string
+	mutators []tag.Mutator
+	measure  *stats.Int64Measure
+}
+
+var (
+	_ workqueue.GaugeMetric   = (*intMetric)(nil)
+	_ workqueue.CounterMetric = (*intMetric)(nil)
+)
+
+// Inc implements CounterMetric
+func (m intMetric) Inc() {
+	stats.RecordWithTags(context.Background(), m.mutators, m.measure.M(1))
```

Comment:
Could you implement something similar to `Record` func in `metrics/record.go`? If we record the data unconditionally, they will be sent to custom metrics domain when Stackdriver is used, which will cause Stackdriver charge.

Reply:
I adapted this method to support options and the code to use it.